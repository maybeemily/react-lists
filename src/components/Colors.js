import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

function Colors({ colors }) {
  const colorList = colors.map(color => {
    return (<li key={color.name} style={{ color: color, fontSize: '2em' }}>
      <Color name={color.name} hex={color.hex} rgb={color.rgb} />
    </li>
    );
  });

  return (
    <ul>
      { colorList }
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};

export default Colors;
