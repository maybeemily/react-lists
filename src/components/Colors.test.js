import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

describe('Colors component', () => {
  it('renders Colors', () => {
    const testColors = [
      { name: 'olive', hex: '#808000', rgb: '(0, 0, 128)' },
      { name: 'olivedrab', hex: '#6B8E23', rgb: '(128, 128, 0)' },
      { name: 'orange', hex: '#FFA500', rgb: '(107, 142, 35)' }
    ];
    const wrapper = shallow(<Colors colors={testColors} />);
    expect(wrapper).toMatchSnapshot();
  });
});
