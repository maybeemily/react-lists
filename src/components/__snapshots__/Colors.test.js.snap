// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component renders Colors 1`] = `
<ul>
  <li
    key="olive"
    style={
      Object {
        "color": Object {
          "hex": "#808000",
          "name": "olive",
          "rgb": "(0, 0, 128)",
        },
        "fontSize": "2em",
      }
    }
  >
    <Color
      hex="#808000"
      name="olive"
      rgb="(0, 0, 128)"
    />
  </li>
  <li
    key="olivedrab"
    style={
      Object {
        "color": Object {
          "hex": "#6B8E23",
          "name": "olivedrab",
          "rgb": "(128, 128, 0)",
        },
        "fontSize": "2em",
      }
    }
  >
    <Color
      hex="#6B8E23"
      name="olivedrab"
      rgb="(128, 128, 0)"
    />
  </li>
  <li
    key="orange"
    style={
      Object {
        "color": Object {
          "hex": "#FFA500",
          "name": "orange",
          "rgb": "(107, 142, 35)",
        },
        "fontSize": "2em",
      }
    }
  >
    <Color
      hex="#FFA500"
      name="orange"
      rgb="(107, 142, 35)"
    />
  </li>
</ul>
`;
